### PR TITLE
Moved ckpt_conversion to maxtext

### PR DESCRIPTION
**Pull request by hengtaoguo**
# Description

This PR introduces following restructure changes:
- Move `MaxText/utils/ckpt_conversion` to `maxtext/checkpoint_conversion`.
- Move `MaxText/utils/ckpt_scripts` to `maxtext/checkpoint_conversion/standalone_scripts`.
- Update all corresponding paths, imports and API references in codes/docs/comments.

# Tests

CI. One-off manual test with conversion/forward_pass_logits_check.
```
# Qwen3-0.6B. 
# to_maxtext
python -m maxtext.checkpoint_conversion.to_maxtext maxtext/configs/base.yml hardware=cpu skip_jax_distributed_system=True hf_access_token=xxx model_name=qwen3-0.6b base_output_directory=gs://hengtaoguo-maxtext-logs/checkpoints/qwen3-0.6b/scanned/2026-02-16-11-35 scan_layers=true

# decode
python -m maxtext.decode maxtext/configs/base.yml model_name=qwen3-0.6b tokenizer_path=Qwen/Qwen3-0.6B tokenizer_type=huggingface hf_access_token=xxx load_parameters_path=gs://hengtaoguo-maxtext-logs/checkpoints/qwen3-0.6b/scanned/2026-02-16-11-35/0/items per_device_batch_size=1 run_name=ht_test max_prefill_predict_length=8 max_target_length=32 steps=1 scan_layers=true prompt=\'Paris\ is\ the\'

# forward_pass_logits_check
python -m tests.utils.forward_pass_logit_checker maxtext/configs/base.yml run_name=ht_test model_name=qwen3-0.6b tokenizer_path=Qwen/Qwen3-0.6B load_parameters_path=gs://hengtaoguo-maxtext-logs/checkpoints/qwen3-0.6b/scanned/2026-02-16-11-35/0/items scan_layers=True max_prefill_predict_length=4 hf_access_token=xxx --run_hf_model=True --hf_model_path=Qwen/Qwen3-0.6B --max_kl_div=0.015
```

Result: [gpaste](https://paste.googleplex.com/6495220557545472)

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by bvandermoon — COMMENTED** [review]
Thanks @hengtaoguo. Could you add `to_hf` and `to_maxtext` directories to [match here](https://github.com/AI-Hypercomputer/maxtext/blob/a3c3fd1af25a52795da44b2a4b0d49a619544e20/RESTRUCTURE.md)?

**Comment by bvandermoon** [comment]
> Thanks @hengtaoguo. Could you add `to_hf` and `to_maxtext` directories to [match here](https://github.com/AI-Hypercomputer/maxtext/blob/a3c3fd1af25a52795da44b2a4b0d49a619544e20/RESTRUCTURE.md)?

Discussed offline, @hengtaoguo will handle this as part of a follow-up PR

**Review by bvandermoon — APPROVED** [review]

**Review by NuojCheng — APPROVED** [review]
